File: Data Structures & Algorithms/task-scheduling/submission-1.js
class Solution {
    /**
     * @param {character[]} tasks
     * @param {number} n
     * @return {number}
     */
    leastInterval(tasks, n) {
        const maxh = new PriorityQueue((a, b) => b[0] - a[0]);
        const map = {};

        for (let t of tasks) {
            map[t] = (map[t] || 0) + 1;
        }

        for (let [key, value] of Object.entries(map)) {
            maxh.enqueue([value, key]);
        }

        const q = new Queue();

        let time = 0;


        while (!maxh.isEmpty() || !q.isEmpty()) {
            time++;
            if (!q.isEmpty() && q.front()[1] === time) {
                const [count, _releaseTime, task] = q.dequeue();
                maxh.enqueue([count, task]);
            }

            if (!maxh.isEmpty()) {
                let [count, task] = maxh.dequeue();
                count--;
                if (count > 0) {

                    const releaseTime = time + n + 1;
                    q.enqueue([count, releaseTime, task]);
                }
            }
        }

        return time;

    }
}

File: Data Structures & Algorithms/decode-ways/submission-1.java
class Solution {
    public int numDecodings(String s) {
        int count = 0;
        char[] sArr = s.toCharArray();

        for(int i = 0; i < sArr.length; i++){
            if(i == 0 && sArr[i] == '0'){
                return 0;
            }

            if(sArr[i] == '0' && (sArr[i - 1] != '1' && sArr[i - 1] != '2')){
                return 0;
            }

            if(sArr[i] == '1'){
                if(i + 1 < sArr.length && Character.getNumericValue(sArr[i + 1]) >= 1 && Character.getNumericValue(sArr[i + 1]) <= 9){
                     count++;
                }  
            }

            if(sArr[i] == '2'&& i + 1 < sArr.length && Character.getNumericValue(sArr[i + 1]) >= 1 && Character.getNumericValue(sArr[i + 1]) <= 6){
                count++;
            }

        }

        count++;

        return count;


    }
}
